### ISSUE TITLE
Using into() in the From trait without having implemented Into should generate a compilation error.

### ISSUE THREAD

**Issue by ThibaultEbroussard** [rust-lang/rust#113531]
### Code

```Rust
fn main() {
    #[derive(Debug)]
    struct Number(f32);

    impl From<f32> for Number {
        fn from(value: f32) -> Self {
            value.into()
        }
    }

    let value: Number = (10.0).into();
    println!("{value:?}");
}
```

### Current output

```Shell
Compiling number v0.1.0 (/home/number)
    Finished dev [unoptimized + debuginfo] target(s) in 0.30s
     Running `target/debug/number`

thread 'main' has overflowed its stack
fatal runtime error: stack overflow
[1]    25988 IOT instruction (core dumped)  cargo run
```

### Desired output

```Shell
The compiler should detect that we are calling Into<Number> for f32 trait which is automatically implemented by From<f32> for Number trait in our implementation of the From<f32> for Number trait.
```

**Comment by riking** [comment]
Implementation would be a deny-by-default lint. Not sure whether to recommend Clippy first or straight into rustc.

**Comment by Nilstrieb** [comment]
I agree that this should live in clippy instead of rustc, @matthiaskrgr do you want to move the issue :)?
